pkg/machine/cleanup.go

```diff
@@ -0,0 +1,66 @@
+package machine
+
+import (
+	"os"
+	"os/signal"
+	"sync"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+type CleanupCallback struct {
+	Funcs []func() error
+	mu    sync.Mutex
+}
+
+func (c *CleanupCallback) Clean(err *error) {
+	// do not remove created files if the init is successful
+	if *err == nil {
+		return
+	}
+
```

Comment:
With the suggestions described below you could additionally reduce some code dup and breakout the suggestion cleanup iteration below thats in CleanOnSignal into a separate function and call it here. For example you could have Clean(), CleanIf(err), and CleanOnSignal() the latter two call Clean() internally